zebra-chain/src/amount.rs

```diff
@@ -324,6 +324,15 @@ where
     }
 }
 
+impl<'amt, C> std::iter::Sum<&'amt Amount<C>> for Result<Amount<C>>
+where
+    C: Constraint + std::marker::Copy + 'amt,
+{
+    fn sum<I: Iterator<Item = &'amt Amount<C>>>(iter: I) -> Self {
+        iter.copied().sum()
+    }
+}
```

Comment:
I'm using `rustc 1.55.0-nightly (7100b311d 2021-07-01)`, it gave me a suggestion for each lifetime I needed to add. Hopefully those suggestions will make it into stable soon.